examples/main/main.cpp

```diff
@@ -360,6 +360,12 @@ int main(int argc, char ** argv) {
                     }
                 }
                 if (i > 0) {
+                    // check if we've used up all the prompt but not all cached tokens
+                    if (embd.size() == i && n_session_consumed < (int) session_tokens.size()) {
+                        // force revaluation of the last token to recalculate logits
+                        i--;
+                        n_past--;
```

Comment:
Might be a dumb question, but if does `i > 0` guarantee that `n_past > 0` as well?

Reply:
Yeah, in the loop just above this, for every `i++` you get an `n_past++`.